airgun/widgets.py

```diff
@@ -1118,6 +1118,47 @@ class EditableEntryCheckbox(EditableEntry):
     edit_field = Checkbox(locator=".//input[@type='checkbox']")
 
 
+class EditableEntryMultiCheckbox(EditableEntry):
+    """Should be used in case :class:`EditableEntry` widget represented not by
+    a field, but by a set of checkboxes.
+    """
+    ITEMS_LOCATOR = ".//p"
+    ITEM_LOCATOR = ".//p/text()[normalize-space(.)='{0}']/preceding-sibling::input"
+
+    def __init__(self, parent, locator=None, name=None, logger=None):
+        if locator and name or not locator and not name:
+            raise TypeError('Please specify either locator or name')
+        locator = (
+            locator or
+            ".//dt[normalize-space(.)='{}']"
+            "/following-sibling::dd[1]".format(name)
+        )
+        super(EditableEntryMultiCheckbox, self).__init__(parent, locator=locator,
+                                                         logger=logger)
+        self.edit_button.click()
+        self.items = {
+            self.browser.text(el).strip():
+            Checkbox(locator=(self.ITEM_LOCATOR.format(self.browser.text(el).strip())))
+            for el in self.browser.elements(self.ITEMS_LOCATOR, parent=self)}
+        self.cancel_button.click()
+
+    def fill(self, value):
+        """Check or uncheck one of the checkboxes
+
+        :param value: string with specification of fields' values
+            Example: value={'details.addons': {'Test addon 1': True, 'Test addon 2': False}}
+        """
+
+        self.edit_button.click()
+
+        for addon in value.keys():
+            field_name = addon
+            field_value = value[addon]
+            self.items[field_name].__get__(self).fill(field_value)
```

Comment:
The item is Checkbox, we should have a function get_item(name)
```python
for item_key, item_value in value:
    item = self.get_item(item_key)
    if not item:
        raise ValueError('a message saying that item with key is not found')
     item.fill(key_value)
```
or I am missing something?

Reply:
"Value" is user input (value which should be set), not the Checkbox's current value. Or maybe I am missing your point.

Reply:
suppose you have this check boxes ['Test Addon1', 'Test Addon2' ]
the user want to set value the user call fill({'Test Addon1': True, 'Test Addon2': False}) 
this check 'Test Addon1' if not checked and uncheck 'Test Addon2' if checked

we pass in the loop the names ['Test Addon1', 'Test Addon2' ] to get_item function to get the corresponding checkboxes (items) that we fill with the corresponding values True or False

Reply:
Maybe the problem is I don't know the get_item function (and can't see it either used nor defined in whole airgun). Is it supposed to return the Checkbox object based on its name? Where is it defined? When I take your code and replace mine with it, I get "object has no attribute 'get_item'" which is understandable because neither EditableEntryMultiCheckbox or its ancestors have the method defined.

Reply:
you have to create that get_item function :)